core/src/main/java/org/ldaptive/pool/BlockingConnectionPool.java

```diff
@@ -235,13 +235,13 @@ public void putConnection(final Connection c)
     logger.trace("waiting on pool lock for check in {}", poolLock.getQueueLength());
     poolLock.lock();
     try {
-      if (active.remove(pc)) {
-        if (valid) {
-          available.add(pc);
-          pc.getPooledConnectionStatistics().addAvailableStat();
-          logger.trace("returned active connection: {}", pc);
-          poolNotEmpty.signal();
-        }
+      if (!valid) {
```

Comment:
I would think you'd want to remove the connection from the pool if the validity check failed implicitly (throws) or explicitly (returns false). It appears that the `Validator` and `Passivator` components that ship with _ldaptive_ do the right thing and return false in the face of exceptions, but the component interface doesn't really speak to what happens when errors occur. Given the importance of the validity check in maintaining pool state, you might want to code defensively here and add some exception handling either around the `validateAndPassivateConnection` call or internally to the method.

Reply:
Pushed a fix to add try-catch blocks.